storops_test/unity/resource/test_snap_schedule.py

```diff
@@ -0,0 +1,443 @@
+         'error_msg': 'days_of_month in UnitySnapScheduleRule should be a '
+                      'list and not an empty list.'},
+        {'schedule_type': ScheduleTypeEnum.DAY_AT_HHMM, 'minute': None,
+         'hours': None, 'days_of_week': None, 'days_of_month': [],
+         'interval': None, 'is_auto_delete': None, 'retention_time': None,
+         'access_type': None,
+         'error_msg': 'days_of_month in UnitySnapScheduleRule should be a '
+                      'list and not an empty list.'},
+        {'schedule_type': ScheduleTypeEnum.DAY_AT_HHMM, 'minute': None,
+         'hours': None, 'days_of_week': None, 'days_of_month': [0, 3],
+         'interval': None, 'is_auto_delete': None, 'retention_time': None,
+         'access_type': None,
+         'error_msg': 'each value of days_of_month in UnitySnapScheduleRule '
+                      'should be \[1, 31\].'},  # noqa
+        {'schedule_type': ScheduleTypeEnum.DAY_AT_HHMM, 'minute': None,
+         'hours': None, 'days_of_week': None, 'days_of_month': [32, 3],
+         'interval': None, 'is_auto_delete': None, 'retention_time': None,
+         'access_type': None,
+         'error_msg': 'each value of days_of_month in UnitySnapScheduleRule '
+                      'should be \[1, 31\].'},  # noqa
+    )
+    @ddt.unpack
+    def test_to_embedded_value_error(self, schedule_type, minute, hours,
+                                     days_of_week, days_of_month, interval,
+                                     is_auto_delete, retention_time,
+                                     access_type, error_msg):
+        assert_that(
+            calling(UnitySnapScheduleRule.to_embedded).with_args(
+                schedule_type, minute=minute, hours=hours,
+                days_of_week=days_of_week, days_of_month=days_of_month,
+                interval=interval, is_auto_delete=is_auto_delete,
+                retention_time=retention_time, access_type=access_type),
+            raises(ValueError, error_msg))
+
+    @ddt.data(
+        {'hour_interval': 3, 'minute': None,
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'expected':
+             {'type': ScheduleTypeEnum.N_HOURS_AT_MM, 'minute': 0,
+              'hours': None, 'daysOfWeek': None,
+              'daysOfMonth': None,
+              'interval': 3, 'isAutoDelete': None, 'retentionTime': None,
+              'accessType': None}},
+        {'hour_interval': 3, 'minute': 30,
+         'is_auto_delete': True, 'retention_time': 3600,
+         'access_type': FilesystemSnapAccessTypeEnum.CHECKPOINT,
+         'expected':
+             {'type': ScheduleTypeEnum.N_HOURS_AT_MM, 'minute': 30,
+              'hours': None, 'daysOfWeek': None,
+              'daysOfMonth': None,
+              'interval': 3, 'isAutoDelete': True, 'retentionTime': 3600,
+              'accessType': FilesystemSnapAccessTypeEnum.CHECKPOINT}},
+    )
+    def test_every_n_hours(self, kwargs):
```

Comment:
No negative test case for every_n_hours.

Reply:
`every_n_hours` and `every_n_days` are just invoking `to_embedded` directly without extra logic. So, testing `to_embedded` with negative cases is enough.